framework/include/systems/NonlinearSystemBase.h

```diff
@@ -873,6 +909,13 @@ class NonlinearSystemBase : public SolverSystem, public PerfGraphInterface
 
   bool _computing_initial_residual;
 
+  /// The pre-SMO residual
+  Real _pre_smo_residual;
+  /// The initial residual
+  Real _initial_residual;
```

Comment:
I named them this way so that there is no conceptual clash between the two residuals. "Initial residual is initial residual".

Reply:
Can you clarify in the doxygen whether this may be the pre-SMO residual or the post-SMO residual?